### PR TITLE
Improve WebDriver error message

### PR DESCRIPTION
**Pull request by jushutch**
The current WebDriver error message is too vague, since it's a generic message with no specific guidance for each browser. Adding a link to the [documentation on WebDrivers](https://flutter.dev/docs/testing/integration-tests#running-in-a-browser)  to help troubleshooting. This same pattern exists elsewhere in the code and works well in directing people to the right information for resolving the error.

An example of the new message including the link:
```Unable to start WebDriver Session for Flutter for Web testing. 
Make sure you have the correct WebDriver Server running at 4444. 
Make sure the WebDriver Server matches option --browser-name. 
For more information see: https://flutter.dev/docs/testing/integration-tests#running-in-a-browser 
SocketException: OS Error: The remote computer refused the network connection.
, errno = 1225, address = localhost, port = 61383
```

This PR fixes issue #78472.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
It seems reasonable that this could be test exempt but I would ask in the discord according to the instructions above.

**Comment by jonahwilliams** [comment]
@Hixie did this get marked as test exempt or not?

**Comment by Hixie** [comment]
It did not (test-exempt PRs have a comment somewhere that starts with `test-exempt: `). I would recommend having a test, in part because hopefully a test would catch the error in the PR. :-)

**Comment by jushutch** [comment]
I just added a simple test to check that the error message contains the link that was added. However I must be missing the error in the PR that was mentioned; @Hixie would you mind elaborating to save some time? Is there more that I should be explicitly testing for in this change? Thanks for the help!

File: packages/flutter_tools/test/general.shard/drive/drive_service_test.dart
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/drive/drive_service_test.dart:382]: This is dangerous because if we accidentally change this to not throw a tool exit the test will still pass. Consider something like this:

```dart
expect(() => driverService.startTest(...), throwsToolExit(message: ...))
```

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart
Comment (Hixie) [inline packages/flutter_tools/lib/src/drive/web_driver_service.dart:115]: the trailing space before the newline here is what i was referring to

File: packages/flutter_tools/test/general.shard/drive/drive_service_test.dart
Comment (jushutch) [reply packages/flutter_tools/test/general.shard/drive/drive_service_test.dart:382]: I agree, that's exactly what I was looking for and couldn't find, thanks!

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart
Comment (jushutch) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart:115]: Okay, thanks for pointing it out. I assumed this was the correct style based on the existing message, but let me know if you prefer otherwise and I can remove the trailing spaces for the entire message.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart:115]: Friendly ping @Hixie

**Review by jonahwilliams — APPROVED** [review]
LGTM if lg to @Hixie

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart
Comment (Hixie) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart:115]: The existing text also needs fixing, I hadn't noticed that.

There shouldn't be a trailing space before a newline.

File: packages/flutter_tools/test/general.shard/drive/drive_service_test.dart
Comment (Hixie) [inline packages/flutter_tools/test/general.shard/drive/drive_service_test.dart:387]: I would probably make this test more strict. It's not just that there's a link that matters, it's also that the text is correct around the link, right?

**Comment by jonahwilliams** [comment]
This branch unfortunately has merge conflicts that need to be handled before we could land it.

File: packages/flutter_tools/test/general.shard/drive/drive_service_test.dart
Comment (jushutch) [reply packages/flutter_tools/test/general.shard/drive/drive_service_test.dart]: Good point, I made the test more strict by including the preamble 👍

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart
Comment (jushutch) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart]: I removed all of the trailing spaces before new lines in this message 👍

**Comment by jonahwilliams** [comment]
I don't know why this got stuck so I'm merging it

**Pull request merged**
